api-channel-issue-tracker/src/main/java/com/synopsys/integration/alert/api/channel/issue/search/IssueTrackerSearcherV2.java

```diff
@@ -0,0 +1,184 @@
+            } else {
+                projectIssueSearchResults.add(searchResult);
+            }
+        }
+        return projectIssueSearchResults;
+    }
+
+    private List<ActionableIssueSearchResult<T>> findIssuesByAllComponents(ProviderDetails providerDetails, LinkableItem project, LinkableItem projectVersion, List<BomComponentDetails> bomComponents) throws AlertException {
+        List<ProjectIssueSearchResult<T>> componentIssues = new LinkedList<>();
+        for (BomComponentDetails bomComponent : bomComponents) {
+            List<ProjectIssueSearchResult<T>> issuesByComponent = projectVersionComponentIssueFinder.findIssuesByComponent(providerDetails, project, projectVersion, bomComponent);
+            componentIssues.addAll(issuesByComponent);
+        }
+        return componentIssues
+                   .stream()
+                   .map(this::convertToUpdateResult)
+                   .collect(Collectors.toList());
+    }
+
+    private ActionableIssueSearchResult<T> findIssueByProjectIssueModel(ProjectIssueModel projectIssueModel) throws AlertException {
+        ExistingIssueDetails<T> existingIssue = null;
+        ItemOperation searchResultOperation = ItemOperation.UPDATE;
+
+        List<ExistingIssueDetails<T>> existingIssues = exactIssueFinder.findExistingIssuesByProjectIssueModel(projectIssueModel);
+        int foundIssuesCount = existingIssues.size();
+
+        if (foundIssuesCount == 1) {
+            existingIssue = existingIssues.get(0);
+
+            Optional<ItemOperation> policyOperation = projectIssueModel.getPolicyDetails().map(IssuePolicyDetails::getOperation);
+            Optional<IssueVulnerabilityDetails> optionalVulnerabilityDetails = projectIssueModel.getVulnerabilityDetails();
+            if (policyOperation.isPresent()) {
+                searchResultOperation = policyOperation.get();
+            } else if (optionalVulnerabilityDetails.isPresent()) {
+                IssueVulnerabilityDetails issueVulnerabilityDetails = optionalVulnerabilityDetails.get();
+                boolean isResolvableOrUnknown = IssueStatus.RESOLVABLE.equals(existingIssue.getIssueStatus()) || IssueStatus.UNKNOWN.equals(existingIssue.getIssueStatus());
+                boolean isReopenableOrUnknown = IssueStatus.REOPENABLE.equals(existingIssue.getIssueStatus()) || IssueStatus.UNKNOWN.equals(existingIssue.getIssueStatus());
+                if (issueVulnerabilityDetails.areAllComponentVulnerabilitiesRemediated() && isResolvableOrUnknown) {
+                    searchResultOperation = ItemOperation.DELETE;
+                } else if (!issueVulnerabilityDetails.areAllComponentVulnerabilitiesRemediated() && isReopenableOrUnknown) {
+                    searchResultOperation = ItemOperation.ADD;
+                } else {
+                    searchResultOperation = ItemOperation.UPDATE;
+                }
+            }
```

Comment:
There's a lot of nesting here. Perhaps we could have a tool or add to projectIssueModel the ability to determine the searchResultOperation?

Reply:
Would be a change for another time, but probably something we will want to simplify in the future